clang/test/SemaSYCL/intel-fpga-uses-global-work-offset.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clang_cc1 -Wno-return-type -fsycl-is-device -fcxx-exceptions -fsyntax-only -ast-dump -verify -pedantic %s | FileCheck %s
+
+struct FuncObj {
+  [[intelfpga::uses_global_work_offset(1)]] void operator()() {}
+};
+
+template <typename name, typename Func>
+void kernel(Func kernelFunc) {
+  kernelFunc();
+}
+
+int main() {
+  // CHECK: SYCLIntelUsesGlobalWorkOffsetAttr{{.*}}Enabled
+  kernel<class test_kernel1>([]() {
+    FuncObj();
+  });
+
+  // CHECK: SYCLIntelUsesGlobalWorkOffsetAttr
+  // CHECK-NOT: Enabled
+  kernel<class test_kernel2>(
```

Comment:
```suggestion
  // expected-no-diagnostics
  kernel<class test_kernel2>(
```

Reply:
This is not valid syntax for semantic check. Clang reports the following: `"expected directive cannot follow 'expected-no-diagnostics' directive"`